apps/game_backend/lib/game_backend/campaigns.ex

```diff
@@ -41,10 +37,88 @@ defmodule GameBackend.Campaigns do
     |> Repo.insert()
   end
 
+  @doc """
+  Inserts a campaign.
+  """
+  def insert_campaign(attrs, opts \\ []) do
+    %Campaign{}
+    |> Campaign.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
+  @doc """
+  Inserts a campaign progress.
+  """
+  def insert_campaign_progress(attrs, opts \\ []) do
+    %CampaignProgress{}
+    |> CampaignProgress.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
+  @doc """
+  Inserts a super campaign.
+  """
+  def insert_super_campaign(attrs, opts \\ []) do
+    %SuperCampaign{}
+    |> SuperCampaign.changeset(attrs)
+    |> Repo.insert(opts)
+  end
+
   @doc """
   Get a level by id.
+
+  Returns `{:error, :not_found}` if no level is found.
   """
   def get_level(level_id) do
-    Repo.get(Level, level_id) |> Repo.preload(units: :items, units: :character)
+    level = Repo.get(Level, level_id) |> Repo.preload(units: :items, units: :character)
+    if level, do: {:ok, level}, else: {:error, :not_found}
+  end
+
+  @doc """
+  Get a campaign progress by user id and campaign id.
+  Returns `{:error, :not_found}` if no progress is found.
+  """
+  def get_campaign_progress(user_id, campaign_id) do
+    campaign_progress =
+      Repo.one(
+        from(cp in CampaignProgress,
+          where: cp.user_id == ^user_id and cp.campaign_id == ^campaign_id,
+          preload: [level: :campaign]
+        )
+      )
+
+    if campaign_progress, do: {:ok, campaign_progress}, else: {:error, :not_found}
+  end
+
+  @doc """
+  Returns what the next level is for a user with its campaign in a tuple.
+
+  Usually it is the level in the same campaign with the next `level_number`.
+  If it doesn't exist it means the campaign is over and we go find the next one by
+  `campaign_number`. If it doesn't exist, we have cleared the SuperCampaign and we return
+  the same given campaign and level instead.
+  """
+  def get_next_level(level) do
+    campaign = level.campaign
+
+    next_level =
+      Repo.get_by(Level, campaign_id: campaign.id, level_number: level.level_number + 1)
+
+    if next_level do
+      {campaign.id, next_level.id}
+    else
+      next_campaign =
+        Repo.get_by(Campaign,
+          super_campaign_id: campaign.super_campaign_id,
+          campaign_number: campaign.campaign_number + 1
+        )
+
+      if next_campaign do
+        first_level = Repo.get_by(Level, campaign_id: next_campaign.id, level_number: 1)
+        {next_campaign.id, first_level.id}
+      else
+        {campaign.id, level.id}
+      end
+    end
```

Comment:
We could avoid these nested if's using a case with 3 statements:
1. has_next_level? ->
2. has_next_campaign? ->
3. true ->